Formula/jackett.rb

```diff
@@ -0,0 +1,52 @@
+class Jackett < Formula
+  desc "API Support for your favorite torrent trackers"
+  homepage "https://github.com/Jackett/Jackett"
+  url "https://github.com/Jackett/Jackett/archive/refs/tags/v0.20.805.tar.gz"
+  sha256 "5b37a01517d2230879fdd0b5f5a3344abcc7718adee9584c2b7b40e5b99b721b"
+  license "GPL-2.0-only"
+  head "https://github.com/Jackett/Jackett.git", branch: "master"
+
+  depends_on "dotnet"
+  depends_on macos: :high_sierra
+
+  def install
+    cd "src" do
+      os = OS.mac? ? "osx" : OS.kernel_name.downcase
+      arch = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch.to_s
+
+      args = %W[
+        --configuration Release
+        --framework net#{Formula["dotnet"].version.major_minor}
+        --output #{libexec}
+        --runtime #{os}-#{arch}
+        --no-self-contained
+        /p:AssemblyVersion=#{version}
+        /p:FileVersion=#{version}
+        /p:InformationalVersion=#{version}
+        /p:Version=#{version}
+      ]
+
+      system "dotnet", "publish", "Jackett.Server", *args
+    end
+
+    (bin/"jackett").write_env_script libexec/"jackett", DOTNET_ROOT: "${DOTNET_ROOT:-#{Formula["dotnet"].opt_libexec}}"
+  end
+
+  def caveats
+    <<~EOS
+      Configuration files: ~/.config/Jackett (or XDG_CONFIG_HOME environment variable, if set)
+    EOS
+  end
+
+  service do
+    run [opt_bin/"jackett", "--NoUpdates"]
+    keep_alive true
+    working_dir libexec
+    log_path var/"log/jackett.log"
+    error_log_path var/"log/jackett.log"
+  end
+
+  test do
+    assert_match(/^Jackett v#{Regexp.escape(version)}$/, shell_output("#{bin}/jackett --version 2>&1; true"))
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.